pkg/svcat/service-catalog/broker.go

```diff
@@ -120,50 +131,79 @@ func (sdk *SDK) Register(brokerName string, url string, opts *RegisterOptions) (
 		}
 
 	}
-	request := &v1beta1.ClusterServiceBroker{
-		ObjectMeta: v1.ObjectMeta{
-			Name: brokerName,
-		},
-		Spec: v1beta1.ClusterServiceBrokerSpec{
-			CommonServiceBrokerSpec: v1beta1.CommonServiceBrokerSpec{
-				CABundle:              caBytes,
-				InsecureSkipTLSVerify: opts.SkipTLS,
-				RelistBehavior:        opts.RelistBehavior,
-				RelistDuration:        opts.RelistDuration,
-				URL:                   url,
-				CatalogRestrictions: &v1beta1.CatalogRestrictions{
-					ServiceClass: opts.ClassRestrictions,
-					ServicePlan:  opts.PlanRestrictions,
-				},
-			},
+	objectMeta := v1.ObjectMeta{Name: brokerName}
+	commonServiceBrokerSpec := v1beta1.CommonServiceBrokerSpec{
+		CABundle:              caBytes,
+		InsecureSkipTLSVerify: opts.SkipTLS,
+		RelistBehavior:        opts.RelistBehavior,
+		RelistDuration:        opts.RelistDuration,
+		URL:                   url,
+		CatalogRestrictions: &v1beta1.CatalogRestrictions{
+			ServiceClass: opts.ClassRestrictions,
+			ServicePlan:  opts.PlanRestrictions,
 		},
 	}
-	if opts.BasicSecret != "" {
-		request.Spec.AuthInfo = &v1beta1.ClusterServiceBrokerAuthInfo{
-			Basic: &v1beta1.ClusterBasicAuthConfig{
+	if scopeOpts.Scope.Matches(ClusterScope) {
+		request := &v1beta1.ClusterServiceBroker{
+			ObjectMeta: objectMeta,
+			Spec: v1beta1.ClusterServiceBrokerSpec{
+				CommonServiceBrokerSpec: commonServiceBrokerSpec,
+			},
+		}
+		request.Spec.AuthInfo = &v1beta1.ClusterServiceBrokerAuthInfo{}
+		if opts.BasicSecret != "" {
+			request.Spec.AuthInfo.Basic = &v1beta1.ClusterBasicAuthConfig{
 				SecretRef: &v1beta1.ObjectReference{
 					Name:      opts.BasicSecret,
 					Namespace: opts.Namespace,
 				},
-			},
-		}
-	} else if opts.BearerSecret != "" {
-		request.Spec.AuthInfo = &v1beta1.ClusterServiceBrokerAuthInfo{
-			Bearer: &v1beta1.ClusterBearerTokenAuthConfig{
+			}
+		} else if opts.BearerSecret != "" {
```

Comment:
Ah OK I see why you did this. I'm going to open a follow-up issue that ClusterServifeBrokerAuthInfo and ServiceBrokerAuthInfo can be consolidated into a single type and see what people think.

Reply:
https://github.com/kubernetes-incubator/service-catalog/issues/2372